activesupport/lib/active_support/cache.rb

```diff
@@ -187,6 +187,10 @@ class << self
           private_constant :DEFAULT_POOL_OPTIONS
 
           def retrieve_pool_options(options)
+            unless options.key?(:pool) || options.key?(:pool_size) || options.key?(:pool_timeout)
+              options[:pool] = true
```

Comment:
It's a bit of extra work, but if we were to introduce a new config for this new default, and flip it in `load_defaults 7.1`, it would probably avoid issues.

WDYT?